Workbooks/Azure Web Application Firewall/WAF monitoring/WAF Monitoring.workbook

```diff
@@ -0,0 +1,831 @@
+            "multiSelect": true,
+            "quote": "'",
+            "delimiter": ",",
+            "query": "AzureDiagnostics\r\n| where Category contains \"FrontdoorWebApplicationFirewallLog\"\r\n| where policy_s == \"sandycookieWAFpolicy\" or policy_s  == \"AmysAPIMwaf\" or policy_s ==\"sandyappgwdcWAFpolicy\"\r\n| project policy_s | distinct  policy_s\r\n",
+            "value": [
+              "sandyappgwdcWAFpolicy"
+            ],
+            "typeSettings": {
+              "additionalResourceOptions": [],
+              "showDefault": false
+            },
+            "queryType": 0,
+            "resourceType": "microsoft.operationalinsights/workspaces"
+          }
+        ],
+        "style": "pills",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces"
+      },
+      "name": "parameters - 2"
+    },
+    {
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "let FakeData = (datatable (Message:string, ruleName_s:string, clientIp_s:string, clientIP_s:string, action_s:string, transactionId_s:string,trackingReference_s:string) [ \"\", \"\", \"\", \"\", \"\", \"\", \"\" ]);\r\nFakeData | union AzureDiagnostics\r\n| where (ResourceType == \"APPLICATIONGATEWAYS\" or ResourceType == \"FRONTDOORS\" or ResourceType == \"PROFILES\" or ResourceType == \"CDNWEBAPPLICATIONFIREWALLPOLICIES\") and (\"{WAFType:label}\" == \"All\" or (ResourceType == \"APPLICATIONGATEWAYS\" and \"{WAFType:label}\" contains \"application gateway\") or (ResourceType == \"FRONTDOORS\" and  \"{WAFType:label}\" contains \"azure front door\") or (ResourceType == \"PROFILES\" and \"{WAFType:label}\" contains \"azure front door premium\") or (ResourceType==\"CDNWEBAPPLICATIONFIREWALLPOLICIES\" and \"{WAFType:label}\" contains \"cdn\")) and (\"{WAF:label}\" == \"All\" or Resource in ({WAF}))and (policy_s in ({WAFPolicy}))\r\n| where Category == \"FrontdoorWebApplicationFirewallLog\" or Category == \"FrontDoorWebApplicationFirewallLog\" or OperationName == \"ApplicationGatewayFirewall\" or Category == \"WebApplicationFirewallLogs\"\r\n| extend Rule = strcat(ruleName_s, Message), ClientIP = strcat(clientIp_s, clientIP_s)\r\n// Application Gateway has Matched, Blocked, Detected : translates to Matched, Block, Log\r\n// Azure Front Door has Matched, Block, Log : translates to Matched, Block, Log\r\n| extend Action = iif(action_s == \"Blocked\", Action = \"Block\", action_s)\r\n| extend Action = iif(Action == \"Detected\", Action = \"Log\", Action)\r\n| summarize number = count() by Action",
+        "size": 3,
+        "showAnalytics": true,
+        "title": "WAF actions filter",
+        "exportFieldName": "series",
+        "exportParameterName": "SelectedAction",
+        "exportDefaultValue": "*",
+        "showExportToExcel": true,
+        "exportToExcelOptions": "all",
+        "queryType": 0,
+        "resourceType": "microsoft.operationalinsights/workspaces",
```

Comment:
like some of the params, this query doesn't specify any workspace and isn't referenceing the parameter?  (i'm presuming you're testing this query inside workbooks ON a logs workspace, in that context it would inherit that workspace as the workspace to use. outside of logs (like on your resource type or in Azure Monitor -> Workbooks view) this template is all errors that you havn't selected   workspace:
![image](https://user-images.githubusercontent.com/10158007/214456499-6c0992c2-b46d-4227-b251-cd40e530308a.png)

Reply:
I tested in Azure Monitor workbooks and I see the data displayed correctly 
![image](https://user-images.githubusercontent.com/49657746/214457839-db754894-46c0-4137-99d8-c5984f8c2d56.png)
![monitoring metrics](https://user-images.githubusercontent.com/49657746/214457918-f1170b15-c377-4204-930f-0685042fef62.jpg)

Reply:
> like some of the params, this query doesn't specify any workspace and isn't referenceing the parameter? (i'm presuming you're testing this query inside workbooks ON a logs workspace, in that context it would inherit that workspace as the workspace to use. outside of logs (like on your resource type or in Azure Monitor -> Workbooks view) this template is all errors that you havn't selected workspace: ![image](https://user-images.githubusercontent.com/10158007/214456499-6c0992c2-b46d-4227-b251-cd40e530308a.png)

Your assumption is correct "_testing this query inside workbooks ON a logs workspace, in that context it would inherit that workspace as the workspace to use_" . There is no "All" option here so it defaults to all.